src/hooks/useAmbassadors.tsx

```diff
@@ -137,6 +138,8 @@ const fallbackAmbassadors: Record<string, Ambassador> =
 
 // Use a context to fetch the ambassadors from the API
 const Context = createContext<Record<string, Ambassador> | null>(null);
+// Seperate context for the refreshing ambassadors
+const RefreshContext = createContext<(() => Promise<void>) | null>(null);
```

Comment:
I think I'd prefer to expose this as a single context/hook, so probably changing the type of the context to be `{ ambassadors, refresh }`?